typescript/utils/src/objects.ts

```diff
@@ -136,3 +136,82 @@ export function stringifyObject(
   }
   return yamlStringify(JSON.parse(json), null, space);
 }
+
+// Function to recursively remove 'address' properties and lowercase string properties
+export function normalizeConfig(obj: any): any {
+  if (Array.isArray(obj)) {
+    return obj.map(normalizeConfig);
+  } else if (obj !== null && typeof obj === 'object') {
+    const newObj: any = {};
+    for (const key in obj) {
+      if (key !== 'address') {
+        newObj[key] = key === 'type' ? obj[key] : normalizeConfig(obj[key]);
+      }
+    }
+    return newObj;
+  } else if (typeof obj === 'string') {
+    return obj.toLowerCase();
+  }
+
+  return obj;
+}
+
+export function configDeepEquals(v1: any, v2: any): boolean {
```

Comment:
we should use https://nodejs.org/api/assert.html#assertdeepequalactual-expected-message

Reply:
NICEEEEEE